projects/angular/src/utils/popover/providers/popover-events.service.ts

```diff
@@ -92,26 +95,30 @@ export class ClrPopoverEventsService implements OnDestroy {
 
   public addClickListener() {
     if (this.outsideClickClose) {
-      this.documentClickListener = this.renderer.listen(this.document, 'click', (event: MouseEvent) => {
-        if (event === this.ignoredEvent) {
-          // Here we ignore the opening click event (w/o this content opens and immediately closes.
-          delete this.ignoredEvent;
-        } else {
-          this.smartOpenService.open = false;
-          // Rather than a complex change to the focus trap I put focus on the element that was clicked
-          const clickedElement = event.target as HTMLElement;
-          clickedElement.focus();
-        }
-      });
+      this.documentClickListener = this.ngZone.runOutsideAngular(() =>
+        this.renderer.listen(this.document, 'click', (event: MouseEvent) => {
+          if (event === this.ignoredEvent) {
+            // Here we ignore the opening click event (w/o this content opens and immediately closes.
+            this.ignoredEvent = null;
```

Comment:
Curious why using `null` instead of `delete` as we used to?

Reply:
Hey, I wouldn't say there's super specific reason for this one, this is one of the old habbits. This was one of the recommendations from the V8 team on using `delete`:
> However, the delete operator will still result in a significant performance hit on property access if a property other than the most recently added property is deleted from the object. So overall we have to continue to recommend against using delete.

I can revert it back to `delete this.ignoredEvent`.